pkg/reconcilermanager/controllers/reposync_controller_test.go

```diff
@@ -1254,64 +1259,99 @@ func TestRepoSyncUpdateCACert(t *testing.T) {
+		repoSync *v1beta1.RepoSync
+	}{
+		"git": {
+			repoSync: repoSyncWithGit(reposyncNs, reposyncName, reposyncRef(gitRevision), reposyncBranch(branch),
+				reposyncSecretType(configsync.AuthNone),
+				reposyncCACert(v1beta1.GitSource, caCertSecret)),
+		},
+		"oci": {
+			repoSync: repoSyncWithOCI(reposyncNs, reposyncName, reposyncOCIAuthType(configsync.AuthNone),
+				reposyncCACert(v1beta1.OciSource, caCertSecret)),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			certSecret := fake.SecretObject(caCertSecret, core.Namespace(tc.repoSync.Namespace))
+			_, _, testReconciler := setupNSReconciler(t, tc.repoSync, certSecret)
 
-	// validation should return an error
-	err := testReconciler.validateCACertSecret(ctx, rs.Namespace, caCertSecret)
-	require.Error(t, err, "Function call should return an error")
-	require.Equal(t, fmt.Sprintf("caCertSecretRef was set, but %s key is not present in %s Secret", CACertSecretKey, caCertSecret), err.Error(), "unexpected function error")
+			ctx := context.Background()
+
+			// validation should return an error
+			err := testReconciler.validateCACertSecret(ctx, tc.repoSync.Namespace, caCertSecret)
+			require.Error(t, err, "Function call should return an error")
+			require.Equal(t, fmt.Sprintf("caCertSecretRef was set, but %s key is not present in %s Secret", CACertSecretKey, caCertSecret), err.Error(), "unexpected function error")
+		})
+	}
 }
 
 func TestRepoSyncWithoutCACertSecret(t *testing.T) {
-	// reposync setup for testing
 	caCertSecret := "foo-secret"
-	rs := repoSyncWithGit(reposyncNs, reposyncName, reposyncRef(gitRevision), reposyncBranch(branch),
-		reposyncSecretType(configsync.AuthToken), reposyncSecretRef(secretName),
-		reposyncCACert(caCertSecret))
-	gitSecret := secretObjWithProxy(t, secretName, GitSecretConfigKeyToken, core.Namespace(rs.Namespace))
-	gitSecret.Data[GitSecretConfigKeyTokenUsername] = []byte("test-user")
+	testCases := map[string]struct {
+		repoSync *v1beta1.RepoSync
+	}{
+		"git": {
+			repoSync: repoSyncWithGit(reposyncNs, reposyncName, reposyncRef(gitRevision), reposyncBranch(branch),
+				reposyncSecretType(configsync.AuthNone),
+				reposyncCACert(v1beta1.GitSource, caCertSecret)),
+		},
+		"oci": {
+			repoSync: repoSyncWithOCI(reposyncNs, reposyncName, reposyncOCIAuthType(configsync.AuthNone),
+				reposyncCACert(v1beta1.OciSource, caCertSecret)),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// no cert secret is setup to trigger not found error
+			_, _, testReconciler := setupNSReconciler(t, tc.repoSync)
 
-	// no cert secret is setup to trigger not found error
-	_, _, testReconciler := setupNSReconciler(t, rs, gitSecret)
-	ctx := context.Background()
+			ctx := context.Background()
+
+			// validation should return a not found error
+			err := testReconciler.validateCACertSecret(ctx, tc.repoSync.Namespace, caCertSecret)
+			require.Error(t, err, "Function call should return an error")
```

Comment:
i see you're just enhancing the existing test, but elsewhere we usually have the with and without error cases in the same Test, with an optional error field in the case map. Then you can do `testutil.AssertEqual(t, tc.expectedError, err)` to validate the error. Then you can use `testutil.EqualError(...)` in the test case to wrap a real error to validate both type and and message. 

- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/kinds/lookup_test.go#L206
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/kinds/lookup_test.go#L173

Reply:
Done